k8sapiserver/k8sapiserver.go

```diff
@@ -40,12 +40,26 @@ import (
 )
 
 // StartAPIServer starts API server, and it make panic when a error happen.
-func StartAPIServer(etcdURL string) (*restclient.Config, func(), error) {
+func StartAPIServer(apiURL string, etcdURL string) (*restclient.Config, func(), error) {
 	h := &APIServerHolder{Initialized: make(chan struct{})}
-	s := httptest.NewServer(http.HandlerFunc(func(w http.ResponseWriter, req *http.Request) {
+	handler := http.HandlerFunc(func(w http.ResponseWriter, req *http.Request) {
 		<-h.Initialized
 		h.M.GenericAPIServer.Handler.ServeHTTP(w, req)
-	}))
+	})
+
+	l, err := net.Listen("tcp", apiURL)
+	if err != nil {
+		return nil, nil, xerrors.Errorf("Address already in used: %w", err)
```

Comment:
nits: 
We should add the information which describes "what we want to do" here instead of "what error we got" because errors returned by `net.Listen` contain "what error we got" like `listen tcp 127.0.0.1:3000: bind: address already in use`. 

```suggestion
		return nil, nil, xerrors.Errorf("announces on the local network address: %w", err)
```